src/OctoshiftCLI.Tests/ado2gh/Commands/ConfigureAutoLinkCommandTests.cs

```diff
@@ -30,15 +32,72 @@ public async Task Happy_Path()
             var githubRepo = "foo-repo";
             var adoOrg = "foo-ado-org";
             var adoTeamProject = "foo-ado-tp";
+            var keyPrefix = "AB#";
+            var urlTemplate = $"https://dev.azure.com/{adoOrg}/{adoTeamProject}/_workitems/edit/<num>/".Replace(" ", "%20");
 
             var mockGithub = new Mock<GithubApi>(null, null);
+            mockGithub.Setup(x => x.GetAutoLinks(It.IsAny<string>(), It.IsAny<string>()))
+                      .Returns(Task.FromResult(new List<AutoLink>()));
             var mockGithubApiFactory = new Mock<GithubApiFactory>(null, null, null);
             mockGithubApiFactory.Setup(m => m.Create()).Returns(mockGithub.Object);
 
             var command = new ConfigureAutoLinkCommand(new Mock<OctoLogger>().Object, mockGithubApiFactory.Object);
             await command.Invoke(githubOrg, githubRepo, adoOrg, adoTeamProject);
 
-            mockGithub.Verify(x => x.AddAutoLink(githubOrg, githubRepo, adoOrg, adoTeamProject));
+            mockGithub.Verify(x => x.DeleteAutoLink(githubOrg, githubRepo, 1), Times.Never);
+            mockGithub.Verify(x => x.AddAutoLink(githubOrg, githubRepo, keyPrefix, urlTemplate));
+        }
+
+        [Fact]
+        public async Task Idempotency_AutoLink_Exists()
+        {
+            var githubOrg = "foo-org";
+            var githubRepo = "foo-repo";
+            var adoOrg = "foo-ado-org";
+            var adoTeamProject = "foo-ado-tp";
+            var keyPrefix = "AB#";
+            var urlTemplate = $"https://dev.azure.com/{adoOrg}/{adoTeamProject}/_workitems/edit/<num>/".Replace(" ", "%20");
+
+            var mockGithub = new Mock<GithubApi>(null, null);
+            mockGithub.Setup(x => x.GetAutoLinks(It.IsAny<string>(), It.IsAny<string>()))
+                      .Returns(Task.FromResult(new List<AutoLink>
+                      {
+                          new AutoLink{Id = 1, KeyPrefix = keyPrefix, UrlTemplate = urlTemplate},
+                      }));
```

Comment:
While I don't think it's wrong but I wouldn't do `Task.FromResult`, it's better to follow one of the 3 [official ways](https://github.com/Moq/moq4/wiki/Quickstart#async-methods) do mock an `async` method.